sessiontxn/isolation/base.go

```diff
@@ -51,6 +52,7 @@ type baseTxnContextProvider struct {
 	infoSchema    infoschema.InfoSchema
 	txn           kv.Transaction
 	isTxnPrepared bool
+	tp            *sessiontxn.EnterNewTxnType
```

Comment:
Why it is a pointer? And the name `enterNewTxnType` may be better because it is not provider's type

Reply:
Done.